### ISSUE TITLE
[Suggestion] - Put money on the line with a duel.

### ISSUE THREAD

**Issue by mibby** [slipcor/pvparena#99]
I think it would be neat to be able to stake money when dueling players. As long as both players have the amount of money in their account, being able to specify a stake in the command such as /pa 1v1 duel [player] [money] would be nice. As well as notify the player that he has been challenged and that amount of money is being staked - so he doesn't blindly accept to lose all his cash.

**Comment by slipcor** [comment]
Atm only the vault module handles money. I might look into interlinking modules, but that's quite some effort :)

**Comment by slipcor** [comment]
you got it, man :) v1.3.2.94

obviously I could not test sooooo good luck! xD

**Comment by mibby** [comment]
When you don't have enough money and try to duel someone, it outputs to the client ```You don't have $<money entered>.``` It is missing the arena prefix.

When you do have enough money and try to duel someone who doesn't, it outputs to the client ```[Duel] <player> doesn't have enough cash!```. This does have the arena prefix.

When you are in the lounge and leave, it does not refund you. Instead it marks the duel as 'won' and rewards the other player the total pot. I personally believe if you leave while still in the lounge, you should be refunded your fee and the other person should be refunded and kicked out as well, as the duel would be 'cancelled'. So the other player isn't stuck in the lounge breaking the arena for whenever the next person tries to duel someone else.

---

Player 1 types ```/pa duel duel Player2 50```.
Player 2 types ```/pa duel accept```.
Both players are brought to the 'duel' arena lounge.
Both players were deducted $50.
Player 2 types ```/pa leave```.
Player 1 is rewarded $100.
Few seconds go by.
Player 1 is auto-readied. Game tries to start without Player 2.
Player 2 gets the output 'You are not part of an arena!'.

Player 1's screen.
![link2](http://i.imgur.com/yBKFPO0.png)

Player 2's screen.
![link3](http://i.imgur.com/jyTjOy6.png)

http://paste.ubuntu.com/16719493/

With forcestart set to false, this doesn't happen. Though I still believe the winner should be decided when you are put in the battlefield, not in the lounge.

**Comment by slipcor** [comment]
Well the debug is based on a broken way of handling this xD let's try again with those builds:

v1.3.2.120 - [Duel, Vault] address github issue #99

Vault is for the message not having the arena prefix, the rest is done in the Duel module

**Comment by mibby** [comment]
Prefix issue confirmed fixed.

Debug of refund issue.
http://paste.ubuntu.com/16933842/

Player 1 challenged player 2 to a duel for $50.
Player 2 accepted.
Both players were charged and entered in lounge.
Player 1 typed /pa leave.
Both players were kicked out of the lounge.
Player 1's screen says he won $50 and that Player 2 left the fight.
Player 2's screen says he won $50 and that Player 1 left the fight.

Shouldn't it state the duel has been cancelled and each player was refunded rather than each leaving and each 'winning'?

This is tested with forcestart false.

**Comment by slipcor** [comment]
v1.3.2.130 - [Vault,Duel] address github issue #99 - properly display "refunding" text when quitting from the lounge

**Comment by mibby** [comment]
Still informs each player that they left the arena and that the respective other person left the arena as well after them. There is no more 'you have won' output and players are rewarded their money back on ```/pa leave``` while in the lounge, but there is no 'refund' message that the duel has been cancelled. 

I reset the lang.yml file.

Debug.
http://paste.ubuntu.com/17080498/

**Comment by slipcor** [comment]
I messed up, here is the proper fix:

v1.3.2.136 - [Duel] properly inform refunding

**Comment by mibby** [comment]
Few things to note.

1) It still states the opposite person left the arena to both players when you type /pa leave.
2) It now states 'Refunding $[amount]!'.
3) Wouldn't it be better to state the duel has been cancelled rather than the person left the fight while in the lounge?
4) With forcestart: true, it tries to start the duel with both players even if they are no longer part of the arena (in the lounge). Shouldn't it stop the runnable when the duel is cancelled? Otherwise both players get the output 'You are not part of an arena!' a few seconds after leaving the lounge.

**Comment by slipcor** [comment]
well but technically someone left the fight so that is the vanilla game output for when someone leaves the game :/

force start is what it is, force start :P but yeah it should probably stop ...

will fix!

**Comment by slipcor** [comment]
Duel v1.3.3.149 is aiming to fix most of this.

* First player leaves, is announced
* Money is exchanged
* Winner is nevertheless forced to exit the arena
* Exceptions based on leaving / quitting player are prevented

**Comment by mibby** [comment]
Mostly everything seems to be fixed, but a few additional problems I noticed.

1) Duel bet total is not being added to the win pot. i.e. /pa <arena> duel <player> 50, both players are charged $50, but the winner does not earn $100. Only $50.
http://paste.ubuntu.com/18631316/

Might be related to this old ticket?
https://github.com/slipcor/pvparena/issues/98

I have the vault module enabled, but I would assume if a bet is defined, to reward the total of the bet rather than the arena's standard win amount.
```
  vault:
    betpot: false
    bettime: 60
    betWinFactor: 1.0
    betWinTeamFactor: 1.0
    betWinPlayerFactor: 1.0
    entryfee: 0
    killreward: 0.0
    minplaytime: 60
    vminplayers: 2
    minbet: 0.0
    maxbet: 0.0
    winFactor: 1.0
    winreward: 10
    winrewardPlayerFactor: 1.0
    reward:
      playerDeath: 0.0
      playerKill: 0.0
      playerScore: 0.0
      trigger: 0.0
      playerWin: 0.0
    winPot: false
    permfactors:
      pa:
        vault:
          none: 1.0
```

2) When the challenger leaves the duel in the lounge, the other player is not kicked out.
 http://paste.ubuntu.com/18631759/

* Player 1 challenges player 2 to a duel.
/pa duel duel <name> 
* No bet amount was set.
* Player 2 accepts the duel.
/pa duel accept
* Player 1 leaves the duel lounge.
* Player 2 was not kicked out.  
* With ```forcestart: true```, it eventually kicked player 2 out of the duel lounge since not enough players were present?

But oddly enough, when you duel someone with an amount set (/pa <arena> duel <player> 50), both players are kicked out when someone leaves in the lounge.
http://paste.ubuntu.com/18631939/

3) When the other person is kicked out of the arena, shouldn't it say the duel has been cancelled rather than they also left the arena, since they are forced out?

**Comment by slipcor** [comment]
1) Should be working with the latest main plugin version now. I am not sure if I changed much but it works for me after I corrected player resetting
2) I changed this in the latest duel version. It is true that for whatever reason, I did check for the amount before kicking (well the kicking was "part of the refunding", maybe that's why?)
3) Added to latest duel module!

It still says they left, because that's just the mechanic :) But now it notifies them about why ;)
